core/src/avm1/globals/system.rs

```diff
@@ -0,0 +1,544 @@
+use crate::avm1::function::Executable;
+use crate::avm1::object::Object;
+use crate::avm1::property::Attribute::{DontDelete, DontEnum, ReadOnly};
+use crate::avm1::return_value::ReturnValue;
+use crate::avm1::{Avm1, Error, ScriptObject, TObject, Value};
+use crate::context::UpdateContext;
+use enumset::{EnumSet, EnumSetType};
+use gc_arena::MutationContext;
+use num_enum::TryFromPrimitive;
+use std::convert::TryFrom;
+
+/// Available type of sandbox for a given SWF
+pub enum SandboxType {
+    Remote,
+    LocalWithFile,
+    LocalWithNetwork,
+    LocalTrusted,
+}
+
+impl SandboxType {
+    pub fn get_sandbox_name(&self) -> &str {
+        match self {
+            SandboxType::Remote => "remote",
+            SandboxType::LocalWithFile => "localWithFile",
+            SandboxType::LocalWithNetwork => "localWithNetwork",
+            SandboxType::LocalTrusted => "localTrusted",
+        }
+    }
+}
+
+/// The available host operating systems
+pub enum OperatingSystem {
+    WindowsXp,
+    Windows2k,
+    WindowsNt,
+    Windows98,
+    Windows95,
+    WindowsCE,
+    Linux,
+    MacOS,
+}
+
+impl OperatingSystem {
+    pub fn get_os_name(&self) -> &str {
+        match self {
+            OperatingSystem::WindowsXp => "Windows XP",
+            OperatingSystem::Windows2k => "Windows 2000",
+            OperatingSystem::WindowsNt => "Windows NT",
+            OperatingSystem::Windows98 => "Windows 98/ME",
+            OperatingSystem::Windows95 => "Windows 95",
+            OperatingSystem::WindowsCE => "Windows CE",
+            OperatingSystem::Linux => "Linux",
+            OperatingSystem::MacOS => "MacOS",
+        }
+    }
```

Comment:
On my Windows 10 machine, `System.capabilities.os` simply returns `Windows`. Not sure what other values are possible?

Reply:
I got these values from the [spec](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/system/Capabilities.html#os), perhaps it was changed at some point?

Reply:
The spec is fairly old (SWFv19 vs. up to v30+ now), so I'm sure it's changed since then. Maybe best to just add an `OperatingSystem::Other`

Reply:
Which version of flash player did you test on, @Herschel? You may be getting `"Windows"` on old versions if they didn't have support for it baked in, and saw that it was an "unknown" version.

Windows 10 pretends to be generic for old apps unless you deliberately tell it you know about windows 10, because of the windows 9 problem.

Reply:
Yep, I was running in FP 11.3 (CS6), which gives "Windows". Flash Player 32 gives me "Windows 8".

Reply:
That'll be it then. "Windows X" is correct if you can detect it, "Windows" if it's unknown.